api/task_get.go

```diff
@@ -6,6 +6,29 @@ import (
 	"github.com/hashicorp/consul-terraform-sync/logging"
 )
 
+func (h *TaskLifeCycleHandler) GetAllTasks(w http.ResponseWriter, r *http.Request) {
+	h.mu.RLock()
+	defer h.mu.RUnlock()
+
+	ctx := r.Context()
+	requestID := requestIDFromContext(ctx)
+	logger := logging.FromContext(r.Context()).Named(getTaskSubsystemName)
+	logger.Trace("get task request")
+
+	// Retrieve task if it exists
+	taskConfigs, err := h.ctrl.Tasks(ctx)
+	if err != nil {
+		logger.Trace("tasks not found", "error", err)
+		sendError(w, r, http.StatusNotFound, err)
```

Comment:
If Tasks() returns an error, it looks like it would be because of a parsing error, so a 500 response feels more appropriate.

https://github.com/hashicorp/consul-terraform-sync/blob/main/controller/server.go#L189

Reply:
yes, that makes sense, it's acceptable to return an empty list